### PR TITLE
Compatibility with XGBoost 2.0

### PR DESCRIPTION
**Pull request by krfricke**

### CODE REVIEW

File: xgboost_ray/sklearn.py [hunk 1]
```diff
@@ -863,8 +849,9 @@ def _ray_fit_preprocess(self, y) -> Callable:
             ):
                 raise ValueError(label_encoding_check_error)
         else:
-            self.classes_ = np.unique(y)
-            self.n_classes_ = len(self.classes_)
+            if XGBOOST_VERSION < Version("2.0.0"):
+                self.classes_ = np.unique(y)
+            self.n_classes_ = len(np.unique(y))
```
Comment (trivialfis) [inline xgboost_ray/sklearn.py:854]:
```suggestion
            self.n_classes_ = len(self.classes_)
```

**Review by trivialfis — COMMENTED** [review]
Feel free to ping me if there's anything I can help.

File: xgboost_ray/sklearn.py (hunk 1 above)
Comment (krfricke) [reply xgboost_ray/sklearn.py:854]: Isn't this recursive? As

```
    @property
    def classes_(self) -> np.ndarray:
        return np.arange(self.n_classes_)
```

**Review by Yard1 — APPROVED** [review]
lgtm

**Pull request merged**